### DISCUSSION TITLE
Are there existing elements / attributes to group notes?

### DISCUSSION THREAD

**Discussion by martha-thomae** [music-encoding/music-encoding#836] [General]
In mensural notation (especially Ars antiqua repertoire and early Italian Trecento), groups of semibreves that form a unit are usually shown by:
- the use of dots of division (that separates these semibreves from the ones following, see purple arrow in image)
- the use of ligatures (symbols that group two or more notes, green arrow below)
- the use of spacing (see the three descending semibreves in the blue box below)

![Screen Shot 2021-07-06 at 10 17 44 PM](https://user-images.githubusercontent.com/13948831/124691652-24e92d80-deaa-11eb-9c27-c83f4142496b.png)

The first two features (dots and ligatures) are easy to encode with MEI. However, the third scenario (indicating that a group of semibreves form a unit by using spacing) is still not possible to encode with MEI.

The element [`<space>`](https://music-encoding.org/guidelines/v4/elements/space.html) is not a good option for this because it is meant to be placeholder used to fill an incomplete measure or layer, which is not the case here.

I was wondering if there are already existing elements / attributes in the MEI specs that will allow to encode this, something that can indicate groups of notes (apart from the `<ligature>` element). I just want to check the pre-existing elements before we start proposing new elements to encode this. 

An old proposal from the Mensural IG was to use a `<mensUnit>` element:
```xml
<mensUnit>
  <note/>
  <note/>
  <note/>
</mensUnit>
```

**Comment by ahankinson** [comment]
Is this of a case of needing to encode the semantics first, and then indicate how it should be rendered later? I presume all three methods are stylistic differences (they mean the same thing, but are simply written differently). In that case, wouldn’t it be better to encode all three things the same way, but with a flag to say that they should be rendered differently.

in other words, aside from the different visual appearance, how is this functionally different from a ligature?

**Reply by martha-thomae** [reply]
All methods mean the same thing in this particular case. But ligatures and dots of division are used in way more cases (there are ligatures that include other notes—breves and longas—and they are not used to indicate anything regarding interpretation of these notes; and dots of division are used after different note values in triple meter, not only semibreves). 

The spacing (I think) is something very particular of Ars antiqua and Italian Trecento semibreves to indicate the grouping of semibreves that are equivalent to a breve. And the actual interpretation of the notes in these groupings varies a lot.

> Aside from the different visual appearance, how is this functionally different from a ligature?

I think that it could be encoded as a ligature, as long as the rendition of the notes is not an actual ligature (one symbol including all the notes in the group).

**Comment by lpugin** [comment]
Would it be acceptable to see the group as a type of `ligature` with no other visual characteristic than spacing? I am thinking of having an attribute on ligature to indicate that. The advantage is that we would not add another grouping element. Considering that there is apparently no clear name to describe such groups, it could be better not to add one.

**Reply by martha-thomae** [reply]
Actually, that sounds like a good option for encoding them to me.
I think these semibreves are tend to be called _coniunturae_ (it happens in Ars antiqua "Petronian semibreves" and "early Trecento semibreves").
I think not adding another grouping element would be good. We could also use the pre-existing `@form` attribute of `<ligature>` with values `obliqua` | `recta` | `conjunturae`(?)

**Reply by martha-thomae** [reply]
Since Karen is more familiar with this repertoire than me, I am tagging her here. @karend27, what do you think about this idea for encoding groups of semibreves in Ars antiqua and early Trecento? Is this appropriate? Are we missing something important here?
I want to have some proposal for the Mensural IG Meeting :)

**Reply by ahankinson** [reply]
Adding a new value to `data.LIGATUREFORM` sounds like a great solution, if that would address the issue!

**Reply by karend27** [reply]
Have just seen this three (due to the discussion in the Mensural Interest Group). I agree that it makes sense to record this as a ligature. This is not exactly the same as the Petronian semibreves, which are grouped by dots. The defining characteristic here is actually the relationship to the underlaid text: these three semibreves are written like this because they correspond to a single syllable of text. They are a specific notational form, as Martha says, known as a conjunctura, and which is also found in plainchant notation (usually directly following another neume or single note shape), and in modal notation. In mensural notation, if these three pitches were underlaid to three text syllables they would be visually written more spaced out and would have a dot following them.

**Reply by karend27** [reply]
Somewhat related though to Martha's question, and which I think may eventually be useful in rendering (and encoding how the notes look in the original manuscript) would be an element that defines visual spacing, and which might or might not intersect with the semantics. For example, in many of the Machaut manuscripts, the visual spacing of the semibreves and minims often is helpful for performers to quickly figure out the tempus and/or prolatio (notes falling within the same breve might be more closely spaced). Sometimes the note grouping can help figure out something which is a little ambiguous in the notation.

**Reply by rettinghaus** [reply]
@karend27 maybe you want to have a look at https://music-encoding.org/guidelines/dev/elements/pad.html

**Reply by martha-thomae** [reply]
Wow! I didn't know about this element, this might be a good solution for our purposes. Thanks for pointing out this option, @rettinghaus!

**Reply by ahankinson** [reply]
What are the semantics of this element? This seems a bit odd.

If you would like to simply visually group notes, with no additional meaning needed, couldn’t you use the `@ho` attribute to move them closer together?

`@ho` opens a pretty big can of worms when it comes to behaviours for automatic layout and rendering but `<pad>` probably opens a much bigger one when it comes to the semantics of the encoding. Is it describing the padding as an object in itself?

**Comment by pe-ro** [comment]
`<pad>` is a relic of existing music representation systems.  It should probably go away.  It would be better, as @ahankinson suggests, to use `@ho` to slide notes, etc. left and right.

`@ho` is a bit of a problem because the result of it depends on when its value is applied; that is, pre- or post-layout.  But I think one could always assume post-automatic layout.  The phrase "programmatically-determined" in the definition of `@ho` is a rather poor way of saying that the offset should be applied after the object's location has been calculated.  Of course, if the offset didn't exist in the notation being encoded or the encoder didn't care where the note ended up, then `@ho` wouldn't be in the encoding anyway.

**Reply by rettinghaus** [reply]
A bigger problem is, that you need to apply `@ho` to all subsequent notes in automatic layout then. Sometimes you just want to add a little space between symbols without making someone believe there is a `<gap>`. 

To give an example: ABC has the possiblity to [typeset extra space](https://abcnotation.com/wiki/abc:standard:v2.2#typesetting_extra_space). So even `<pad>` carries no semantics I think it makes sense to have it.
